### PR TITLE
Remove OBJECTS_REJ and clean up core render func

### PR DESCRIPTION
**Pull request by FazanaJ**
Completely purged it. I've made a few changes to the function for the sake of cleanliness, but also efficiency, and I fixed the crash caused by debug boxes.

### CODE REVIEW

File: VERSION.txt [hunk 1]
```diff
@@ -1 +1 @@
-v2.1.1
+v2.1.2
```
Comment (someone2639) [inline VERSION.txt:1]: version meme

**Review by someone2639 — COMMENTED** [review]
Since this is 2.2 according to the board the only changes necessary would be to merge the latest changes then its ready to go

File: src/game/rendering_graph_node.c [hunk 2]
```diff
@@ -289,46 +224,17 @@ static struct RenderPhase sRenderPhases[] = {
         .startLayer = LAYER_OCCLUDE_SILHOUETTE_FIRST,
         .endLayer   = LAYER_LAST,
     },
-
- #else
+#else
     // No silhouette, no .rej
```
Comment (gheskett) [inline src/game/rendering_graph_node.c:228]: .rej comment

File: src/game/rendering_graph_node.c [hunk 3]
```diff
@@ -195,75 +195,10 @@ static const Gfx dl_silhouette_end[] = {
...
-    [RENDER_PHASE_REJ_NON_ZB]              = {
-        .startLayer = LAYER_NON_ZB_FIRST,
-        .endLayer   = LAYER_LAST,
-        .ucode      = GRAPH_NODE_UCODE_REJ
-    },
- #endif
-#else
- #if SILHOUETTE
+#if SILHOUETTE
     // Silhouette, no .rej
```
Comment (gheskett) [inline src/game/rendering_graph_node.c:202]: .rej comment

File: src/game/rendering_graph_node.c [hunk 4]
```diff
@@ -427,21 +331,16 @@ void geo_process_master_list_sub(struct GraphNodeMasterList *node) {
...
+        gSPClearGeometryMode(gfx++, G_ZBUFFER);
 #ifdef VISUAL_DEBUG
-    if ( hitboxView) render_debug_boxes(DEBUG_UCODE_DEFAULT | DEBUG_BOX_CLEAR);
-    // Load the world scale identity matrix
-    gSPMatrix(gDisplayListHead++, &identityMatrixWorldScale, G_MTX_MODELVIEW | G_MTX_LOAD | G_MTX_NOPUSH);
-    if (surfaceView) visual_surface_loop();
+        // Load the world scale identity matrix
+        gSPMatrix(gfx++, &identityMatrixWorldScale, G_MTX_MODELVIEW | G_MTX_LOAD | G_MTX_NOPUSH);
+        if (surfaceView) visual_surface_loop(&gfx);
+        render_debug_boxes(&gfx);
```
Comment (gheskett) [inline src/game/rendering_graph_node.c:340]: What happened to `hitboxView`?

File: src/game/rendering_graph_node.h [hunk 5]
```diff
@@ -79,13 +56,12 @@ struct RenderModeContainer {
     };
     #define RENDER_PHASE_SILHOUETTE RENDER_PHASE_ZEX_SILHOUETTE
     #define RENDER_PHASE_NON_SILHOUETTE RENDER_PHASE_ZEX_NON_SILHOUETTE
- #else
+#else
     // No silhouette, no .rej
```
Comment (gheskett) [inline src/game/rendering_graph_node.h:60]: .rej comment

**Review by gheskett — CHANGES_REQUESTED** [review]
This entire branch appears to be based off 2.1.2. This branch needs to be restarted/rebased to be founded off of `develop/2.2.0` in order to be considered for merge.

**Comment by gheskett** [comment]
@FazanaJ please update this thanks

**Comment by gheskett** [comment]
Moved to #735

**Pull request closed without merging**